sugar_test.go

```diff
@@ -44,6 +44,13 @@ func TestSugarWith(t *testing.T) {
 		{[]interface{}{}, []zapcore.Field{}, nil},
 		{[]interface{}{"should ignore"}, []zapcore.Field{}, []observer.LoggedEntry{ignored}},
 		{[]interface{}{"foo", 42, "true", "bar"}, []zapcore.Field{Int("foo", 42), String("true", "bar")}, nil},
+		{[]interface{}{Int("foo", 42)}, []zapcore.Field{Int("foo", 42)}, nil},
+		{[]interface{}{Int("foo", 42), "should ignore"}, []zapcore.Field{Int("foo", 42)}, []observer.LoggedEntry{ignored}},
+		{
+			[]interface{}{"first", "field", Int("foo", 42), "baz", "quux", "should ignore"},
+			[]zapcore.Field{String("first", "field"), Int("foo", 42), String("baz", "quux")},
+			[]observer.LoggedEntry{ignored},
```

Comment:
Added a description to each scenario and switched to keyed literals.

The tests cover both the production and development paths, since they just assert that we're logging at `DPanicLevel`. The tests for `DPanicLevel` verify that we're panicing and erroring appropriately.